state-sync/src/coordinator.rs

```diff
@@ -1058,15 +1054,29 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
         })
     }
 
+    /// Logs the highest seen ledger info version based on the current syncing mode.
+    fn log_highest_seen_version(&self, new_highest_li: Option<LedgerInfoWithSignatures>) {
+        let highest_version = if !self.is_initialized() {
+            self.waypoint.version()
+        } else if let Some(sync_request) = self.sync_request.as_ref() {
+            sync_request.target.ledger_info().version()
+        } else if let Some(new_highest_li) = new_highest_li.as_ref() {
+            new_highest_li.ledger_info().version()
+        } else if let Some(target_ledger_info) = self.target_ledger_info.as_ref() {
+            target_ledger_info.ledger_info().version()
+        } else {
+            self.local_state.synced_version()
+        };
```

Comment:
we aren't comparing this to the previous highest seen li, I don't know if we can read from counters or we need to set a local counter.

Reply:
I've read from the global counter. While I'm not a big fan of this approach as a whole (e.g., a single metric of the highest we've ever seen), this should hopefully achieve what you're asking for? Thoughts?